libs/spark/src/NavBarItem.tsx

```diff
@@ -4,13 +4,21 @@ import {
   Button,
   ButtonBase,
   ButtonBaseProps as MuiButtonBaseProps,
+  ButtonProps,
 } from '@material-ui/core';
-import { ButtonProps } from './Button';
 import { theme } from './styles';
-// import clsx from 'clsx';
-// import { capitalize } from './utils';
 
-export type NavBarItemProps = ButtonProps;
+export interface NavBarItemProps
+  extends Omit<ButtonProps, 'variant' | 'color' | 'size'> {
+  startIcon?: React.ReactNode;
+  endIcon?: React.ReactNode;
+  children?: React.ReactNode;
+  disabled?: boolean;
+  disableElevation?: boolean;
+  disableFocusRipple?: boolean;
+  fullWidth?: boolean;
+  href?: string;
+}
```

Comment:
You don't need to redeclare ones you want. You also don't need to define href, a consumer can do

```
<NavBarItem component="a" href="..." />
```
and that will fine by Mui's default implementation. Likewise for react-router
```
<NavBarItem component={RouterLink} to="..." />
```

See https://material-ui.com/components/buttons/#third-party-routing-library

Reply:
Thanks, William, that's good to know about the React Router implementation. I will need to put some custom props in this interface though, correct? Maybe just children?

Reply:
Without any, I'm getting an empty interface warning.

Reply:
Two Issues:
1. I'm still getting an error message when trying to call `component={Link}`
2. Also trying to work out how to target the clicked state of the button. I attempted using the `aria-current='true'` in styles like Joshua had previously, but was not successful as of yet.

Reply:
I wouldn't add react-router and demonstrate the integration. It's more likely to cause problems than anything. As long as we're extending Button, then we can absorb Mui's test suite and be assured that the integration will still work. I'd recommend just showing `<NavBarItem component="a" href="..." />` use case.

You won't need to add children, again that's already defined. Yeah I'm struggling how to set it equal to the other interface. Maybe ask around

Reply:
What's the harm of doing a type here? I'm not getting an error when using a type when using Omit

```
export type NavBarItemProps = Omit<ButtonProps, 'variant' | 'color' | 'size'>;
// export interface NavBarItemProps
//   extends Omit<ButtonProps, 'variant' | 'color' | 'size'> {}
```

Reply:
I'm not sure. One consequence is that a consumer can extend a type, but they can extend an interface. I know they can union a type and all that jazz, but I feel like there must be a way to do it with interfaces. Maybe just setting the component type to be `ButtonProps`?

Reply:
I haven't solved this piece but I have made a lot of progress on matching the Figma design today. Curious to hear your thoughts on its current state. I'll keep researching on extending interfaces in the meantime.